dhctl/pkg/operations/common.go

```diff
@@ -0,0 +1,36 @@
+package operations
+
+import (
+	"errors"
+	"github.com/deckhouse/deckhouse/dhctl/pkg/state"
+)
+
+type DhctlState map[string][]byte
+
+type OnPhaseFunc func(completedPhase OperationPhase, completedPhaseState DhctlState, nextPhase OperationPhase, nextPhaseCritical bool) error
+
+type OperationPhase string
+
+const (
+	BaseInfraPhase             OperationPhase = "BaseInfra"
+	ExecuteBashibleBundlePhase OperationPhase = "ExecuteBashibleBundle"
+	InstallDeckhousePhase      OperationPhase = "InstallDeckhouse"
+	CreateResourcesPhase       OperationPhase = "CreateResources"
```

Comment:
I think we need one more phase - installAdditionalMastersAndStaticNodes

Reply:
After install deckhouse

Reply:
I would propose to add such phase in the separate PR on top of existing. BTW in the last PR there is small refactor and introduction of PhasedExecutionContext, which is related to the phases.